### PR TITLE
Add feature spec for creating music snippets

### PR DESCRIPTION
**Pull request by entcheva**
TO DO:
- [x]  rebuild user association on snippet creation
- [x]  fix failing `Missing host to link to!` spec

### CODE REVIEW

File: spec/features/user_creates_snippet_spec.rb [hunk 1]
```diff
@@ -0,0 +1,30 @@
+require "rails_helper"
+require "spec_helper"
```
Comment (Euraldius) [inline spec/features/user_creates_snippet_spec.rb]: You don't need to require `spec_helper` ~ it's already required by `rails_helper`

File: app/controllers/snippets_controller.rb [hunk 2]
```diff
@@ -6,8 +6,11 @@ def new
   def create
     @snippet = Snippet.new(snippet_params)
     @snippet.user_id = current_user.id
-    @snippet.save
-    redirect_to dashboard_path
+    if @snippet.save
+      redirect_to dashboard_path
+    else
+      redirect_to new_snippet_path, notice: "Please fill out all fields."
```
Comment (Euraldius) [inline app/controllers/snippets_controller.rb]: You could use the errors from the `Snippet` model to create a more detailed error message

File: spec/factories/clearance.rb [hunk 3]
```diff
@@ -1,10 +1,10 @@
 FactoryGirl.define do
   sequence :email do |n|
-    "user#{n}@example.com"
+    "stardude44#{n}@example.com"
```
Comment (Euraldius) [inline spec/factories/clearance.rb]: 🌟

File: config/environments/test.rb [hunk 4]
```diff
@@ -39,4 +39,8 @@
 
   # Raises error for missing translations
   # config.action_view.raise_on_missing_translations = true
+
+  # Clearance
+  config.action_mailer.default_url_options = { :host => "test.locahost://3000" }
```
Comment (Euraldius) [inline config/environments/test.rb]: Prefer Ruby > 1.9 hash syntax? `host: "test.localhost://3000"`

Why the `//` before `3000`?

(This comment applies to the other config files as well)

File: spec/rails_helper.rb [hunk 5]
```diff
@@ -5,6 +5,7 @@
 
 require 'spec_helper'
 require 'rspec/rails'
+require "clearance/rspec"
```
Comment (entcheva) [inline spec/rails_helper.rb]: Nah dawg

File: app/controllers/snippets_controller.rb (hunk 2 above)
Comment (entcheva) [inline app/controllers/snippets_controller.rb]: How about using `@snippet.errors.messages`? Not sure if that's what you had in mind or if there's a better way.

File: spec/rails_helper.rb (hunk 5 above)
Comment (Euraldius) [inline spec/rails_helper.rb]: It'd be worth adding your own rubicon file so Hound doesn't keep doing this 🐕

Comment (Euraldius) [inline spec/rails_helper.rb]: Oh you did. SWEET 🍭

**Comment by Euraldius** [comment]
Looks good! Remember to squash before you merge 🦑

**Pull request merged**